src/cpptoolsconfigprovider.ts

```diff
@@ -0,0 +1,120 @@
+import path = require("path");
+import * as vscode from "vscode";
+import * as cpptools from "vscode-cpptools";
+import { getMesonBuildOptions, getMesonCompilers, getMesonDependencies } from "./introspection";
+var fs = require("fs");
+
+export class CpptoolsProvider implements cpptools.CustomConfigurationProvider {
+  cppToolsAPI?: cpptools.CppToolsApi;
+  private buildDir: string = "";
+
+  constructor(buildDir: string) {
+    this.buildDir = buildDir;
+  }
+
+  name = "Meson build";
```

Comment:
Let's make this "Meson Build"

Reply:
Make these private

Reply:
Renaming to "Meson Build". However, name and extensionId have to remain public because this class `implements cpptools.CustomConfigurationProvider`

Reply:
Ah Thanks for the explanation!